src/components/EmployHoursApproveBody/EmployHoursApproveBody.jsx

```diff
@@ -0,0 +1,141 @@
+    };
+    const approveReport = (index) => {
+        changeEmployees({"employeeIndex" : employeeIndex, "reportIndexs" : [index], "approval" : "1"})
+    };
+
+    const handleChange = (e) =>  {
+        if(e.target.checked === true){
+            const checkedReportsClone = [...checkedReports];
+            checkedReportsClone[e.target.value] = true;
+            setCheckedReports(checkedReportsClone)
+        }else{
+            if(selectAllReports) setSelectAllReports(!selectAllReports);
+            const checkedReportsClone = [...checkedReports];
+            checkedReportsClone[e.target.value] = false;
+            setCheckedReports(checkedReportsClone)
+       }
+    }
+    const reportsRows = employee.reports.map((report, index) =>{
+        let curseName="-";
+        if(report.courseid != null){
+            curseName = employee.reportingPerimeter[report.projectid].courses.filter(course => course.courseid === report.courseid)[0].courseName;
+        }
+        let subjectName="-";
+        if(report.actionid != null){
+            subjectName = employee.reportingPerimeter[report.projectid].subjects.filter(subject => subject.reportsubjectid === report.actionid)[0].subject;
+        }
+        const projectName = employee.reportingPerimeter[report.projectid].projectName;
+        return(
+            <div className="report-row-container" key={report.reportid}>
+             <div className="report-row-buttons">
+                 <div className="disapprove-container">
+                     <div>דחה</div>
+                     <div onClick={()=>disapproveReport(index)} className={report.approval ==="-1" ? "disapprove-button selected" : "disapprove-button"}>
+                         {report.approval ==="-1"?
+                         <div className="disapprove-button selected inside"></div> :null}
+                     </div>
+                 </div>
+                 <div className="pending-container">
+                     <div>ממתין</div>
+                     <div onClick={()=>pendingReport(index)}className={report.approval === "0" ? "pending-button selected" : "pending-button"}>
+                     {report.approval ==="0"?
+                         <div className="pending-button selected inside"></div> :null}
+                     </div>
+                 </div>
+                 <div className="approve-container">
+                     <div>אשר</div>
+                     <div onClick={()=>approveReport(index)} className={report.approval === "1" ? "approve-button selected" : "approve-button"}>
+                     {report.approval ==="1"?
+                         <div className="approve-button selected inside"></div> :null}
+                     </div>
+                 </div>
+             </div>
+             <div className={report.approval ==="-1" ? "report-row-data red" : report.approval ==="0"? "report-row-data yellow" : "report-row-data green"}>
```

Comment:
don't repeat report-row-data
Also for class names use functional and not implementational names. instead of red use 'disapprove'